MovingBlocks/Terasology
# refactor: replace usage of ThreadManagerSubsystem with reactor Scheduler

This replaces the ThreadManagerSubsystem with Scheduler provided by rxjava. pretty straightforward  since this subsystem is only ever used in the core engine and not to an extent that would make sense. the threads allocated are mostly unused and sit idle. 

toTest:
- verify that game previews are saved
- verify that a screen shot could be taken with F12
- game startup is dependent on some of this threading. 

Contributes to https://github.com/MovingBlocks/Terasology/issues/4798
